src/modules/settings/BlockedSkills.svelte

```diff
@@ -0,0 +1,226 @@
+    if (l.name === list.name) {
+      return false;
+    }
+    if (l.skills.length !== list.skills.length) {
+      return false;
+    }
+    return l.skills.every((s, i) => s === list.skills[i]);
+  });
+}
+
+const loadListChecks = [
+  [
+    (list) => !list,
+    (list) => `Unable to find list "${list.name}". Try refreshing the page.`,
+  ],
+];
+
+const updateListChecks = [
+  [
+    (list) => !blockedSkillLists.some((l) => l.name === list.name),
+    (list) => `Cannot find list named "${list.name}". Try refreshing the page.`,
+  ],
+  [
+    (list) => checkForDuplicates(list),
+    () => 'You already have a list with these skills.',
+  ],
+];
+
+const newListChecks = [
+  [
+    (list) => !list.name.length,
+    () => 'You must name your list.',
+  ],
+  [
+    (list) => list.name.length > 50,
+    () => 'List names can only be at most 50 characters long.',
+  ],
+  [
+    (list) => blockedSkillLists.some((l) => l.name === list.name),
+    (list) => `You already have a list named ${list.name}.`,
+  ],
+  [
+    (list) => checkForDuplicates(list),
+    () => 'You already have a list with these skills.',
+  ],
+];
+
+function checkForErrors(checks, ...args) {
+  const errPair = checks.find(([test]) => test(...args));
+  if (errPair) {
+    error = errPair[1](...args);
+    return true;
+  }
+  return false;
+}
+
+function clearMessages() {
+  error = false;
+}
+
+function storeBlockedSkillsLists() {
+  setValue('blockedSkillLists', JSON.stringify(blockedSkillLists));
+}
+
+function applyList() {
+  clearMessages();
+
+  if (!listName) {
+    return;
+  }
+
+  const skillList = blockedSkillLists.find((l) => l.name === listName);
+
+  if (checkForErrors(loadListChecks, skillList)) {
+    return;
+  }
+
+  clearCheckedSkills();
+  skillList.skills.forEach(checkSkill);
+  submitSkillChanges();
+}
+
+function deleteList() {
+  clearMessages();
+  blockedSkillLists = blockedSkillLists.filter((l) => l.name !== listName);
+  storeBlockedSkillsLists();
+  listName = '';
+}
+
+function createList() {
+  clearMessages();
+
+  const cleanListName = newListName.trim();
+  const newList = {
+    name: cleanListName,
+    skills: getCheckedSkills(),
+  };
+
+  if (checkForErrors(newListChecks, newList)) {
+    return;
+  }
+  blockedSkillLists = [...blockedSkillLists, newList];
+  listName = newList.name;
+  newListName = '';
+  storeBlockedSkillsLists();
+
+  submitSkillChanges();
+}
+
+function updateList() {
+  clearMessages();
+  if (!listName) {
+    return;
+  }
+
+  const skillList = {
+    name: listName,
+    skills: getCheckedSkills(),
+  };
+
+  if (checkForErrors(updateListChecks, skillList)) {
+    return;
+  }
+
+  const index = blockedSkillLists.findIndex((l) => l.name === listName);
+  blockedSkillLists[index] = skillList;
+  storeBlockedSkillsLists();
+  submitSkillChanges();
+}
+
+function findLoadList() {
+  const skills = getCheckedSkills();
+  const list = blockedSkillLists.find((l) => {
+    if (l.skills.length !== skills.length) {
```

Comment:
the conditional here should be a filter
also the whole thing can be refactored into a function
```const list = findList(blockedSkillLists, skills);```
then it can be an import